src/Aeon/Calendar/Gregorian/Day.php

```diff
@@ -50,6 +50,11 @@ public static function fromString(string $date) : self
         return self::fromDateTime(new \DateTimeImmutable($date));
     }
 
+    public static function daysBetween(self $from, self $to) : int
+    {
+        return $from->midnight(TimeZone::UTC())->distanceSince($to->midnight(TimeZone::UTC()))->inDays();
```

Comment:
I like the idea of calculating distance between midnights. It makes total sense! However, I'm not convinced by the static method that returns an integer. It's also possible that one might pass arguments in a different order, which might lead to weird results.
What would you say about changing that method's signature to something that is a bit more object-oriented?

```php
public static distanceUntil(self $day) : TimeUnit;
public static distanceUntil(self $day) : TimeUnit;
```

This way, we would return `TimeUnit`, which is the first-class citizen in aeon libraries and it would make no difference what Day you pass into those methods, they will always give you predictable results. 

Sounds good?

Reply:
In my opinion it is better to implement it as a non-static method that we can chain it,
```php
public function timeBetween(self $day) : TimeUnit;
```
This is more readable and we can use it like this:
```php
$day1->timeBetween($day2)->inDays();
```

Reply:
awesome! `timeBetween` works even better than `distanceUntil`, 👍

Reply:
I made it absolute to avoid minus results because time is a unit that can not be below zero.